src/components/TransitionStaggered/src/TransitionStaggered.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+	<transition-group
+		v-bind="$attrs"
+		tag="div"
+		@enter="handleEnter"
+		@leave="handleLeave"
+	>
+		<slot
+			:data-load-index="loadIndex"
+		/>
+	</transition-group>
+</template>
+
+<script>
+import { staggeredFloatUpFn, floatDownFn } from '@square/maker/utils/transitions';
+
+const defaultItemCount = [
+	{
+		minWidth: 0,
+		itemCount: 1,
+	},
+];
+
+export default {
+	props: {
+		// Number of items to stagger at a time for a viewport
+		// E.g. [
+		// 	{
+		// 		minWidth: 0,
+		// 		staggerItemCount: 1,
+		// 	},
+		// 	{
+		// 		minWidth: 840,
+		// 		staggerItemCount: 4,
+		// 	},
+		// ]
+		staggerItemCount: {
+			type: Array,
+			default: () => defaultItemCount,
+			validator: (staggerItemCount) => {
+				// cannot be empty
+				if (staggerItemCount.length === 0) {
+					return false;
+				}
+				// must have default catch-all count
+				if (staggerItemCount[0].minWidth !== 0) {
+					return false;
+				}
+				return staggerItemCount.every((itemCount) => (itemCount.minWidth
+					|| itemCount.minWidth === 0)
+					&& itemCount.itemCount);
+			},
+		},
+	},
+
+	data() {
+		return {
+			viewportWidth: 0,
+		};
+	},
+
+	mounted() {
+		this.viewportWidth = window.innerWidth;
+	},
+
+	methods: {
+		handleEnter(element, onComplete) {
+			staggeredFloatUpFn({ element, onComplete });
+		},
+
+		handleLeave(element, onComplete) {
+			floatDownFn({ element, onComplete });
+		},
+
+		loadIndex(index) {
+			const { staggerItemCount, viewportWidth } = this;
+			let items;
+			staggerItemCount.forEach((count) => {
+				if (count.minWidth < viewportWidth) {
+					items = count.itemCount;
+				}
+			});
+			if (items === defaultItemCount[0].itemCount) {
+				return index;
+			}
+			const remainder = index % items;
```

Comment:
Posted a design suggestion in [Slack](https://square.slack.com/archives/C0255HML1PD/p1637192756102800) to consider row based staggering on desktop. This can be tested locally by updating the return value to: `Math.floor((index - 1) / items);`
![stagger-row](https://user-images.githubusercontent.com/51469/142308160-4f8d6b77-9c99-401d-81a1-056c285a2724.gif)